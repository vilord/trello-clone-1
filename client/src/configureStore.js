import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import reducer from './reducers';

const middleware = [thunk];
if (process.env.NODE_ENV !== 'production') {
  middleware.push(createLogger());
}

export default preloadedState => {
  return createStore(
    reducer,
    preloadedState,
    composeWithDevTools(applyMiddleware(...middleware)),
  );
};
