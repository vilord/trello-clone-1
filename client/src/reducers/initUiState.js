export default {
  fetching: {
    signup: false,
    login: false,
    activity: false,
  },
  error: {
    kind: '',
    header: '',
    message: '',
  },
};
