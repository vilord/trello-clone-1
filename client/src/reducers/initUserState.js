export default {
  name: '',
  username: '',
  email: '',
  bio: '',
  avatar: '',
  boards: [],
  teams: [],
  assigned_cards: [],
  activity: [],
};
