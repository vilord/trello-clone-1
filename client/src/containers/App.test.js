import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import initUserState from '../reducers/initUserState';
import initUiState from '../reducers/initUiState';

import ConnectedApp, { App } from './App';

describe('App Component', () => {
  const minProps = {
    fetching: false,
    user: initUserState,
    getSession: jest.fn(),
  };

  describe('shallow', () => {
    let wrapper;
    let spy;

    beforeEach(() => {
      wrapper = shallow(<App {...minProps} />);
      spy = jest.spyOn(App.prototype, 'componentDidMount');
    })

    it('renders', () => {
      expect(wrapper.exists()).toBe(true);
      expect(wrapper.length).toEqual(1);
    });

    it('has className App', () => {
      expect(wrapper.hasClass('App')).toBe(true);
    });

    it('calls componentDidMount', () => {
      expect(spy).toHaveBeenCalled();
    });

    it('calls getSession', () => {
      expect(minProps.getSession).toHaveBeenCalled();
    })

    it('does not has App className when fetching', () => {
      expect(wrapper.hasClass('App')).toBe(true);
      wrapper.setProps({ fetching: true });
      expect(wrapper.hasClass('App')).toBe(false);
    });
  });

  describe('Shallow Connected App', () => {
    const initState = {
      user: initUserState,
      ui: initUiState,
    };

    const mockStore = configureStore();
    let store, container;

    beforeEach(() => {
      store = mockStore(initState);
      container = shallow(<ConnectedApp store={store} />);
    });

    it('+++ render the connected(SMART) component', () => {
      expect(container).toHaveLength(1);
    });

    it('+++ check Prop matches with initState', () => {
      expect(container.prop('fetching')).toEqual(initUiState.fetching.login);
      expect(container.prop('user')).toEqual(initState.user);
    });
  });

  xdescribe('>>>H O M E --- REACT-REDUX (Mount + wrapping in <Provider>)', () => {
    const initState = {
      user: initUserState,
      ui: initUiState,
    };

    const middleware = [thunk];
    const mockStore = configureStore(middleware);
    let store, wrapper, spy;

    beforeEach(() => {
      store = mockStore(initState);
      wrapper = mount(
        <Provider store={store}>
          <ConnectedApp />
        </Provider>,
      );
      spy = jest.spyOn(App.prototype, 'componentDidMount');
    });

    it('+++ render the connected(SMART) component', () => {
      expect(wrapper.find(ConnectedApp)).toHaveLength(1);
    });

    it('calls componentDidMount', () => {
      expect(spy).toHaveBeenCalled();
    });

    it('does not render app when fetching the user', () => {
      wrapper.setProps({ fetching: true });
      expect(wrapper.hasClass('App')).toBe(false);
    });

    it('+++ check Prop matches with initState', () => {
      expect(wrapper.find(App).prop('fetching')).toEqual(
        initUiState.fetching.login,
      );
    });

    // it('+++ check action on dispatching ', () => {
    //   return store.dispatch(fetchPosts('programming')).then(() => {
    //     const actions = store.getActions();
    //     expect(actions[2].type).toEqual('RECEIVE_POSTS');
    //   });
    // });
  });

  // Snapshot for App React Component
  describe('snapshot', () => {
    it('mathes snapshot', () => {
      const props = {
        fetching: false,
        user: initUserState,
        getSession: jest.fn(),
      };

      const renderedValue = renderer.create(<App {...props} />).toJSON();
      expect(renderedValue).toMatchSnapshot();
    });
  });
});
