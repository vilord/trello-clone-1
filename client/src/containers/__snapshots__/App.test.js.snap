// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component snapshot mathes snapshot 1`] = `
<div
  className="App"
>
  {"name":"","username":"","email":"","bio":"","avatar":"","boards":[],"teams":[],"assigned_cards":[],"activity":[]}
</div>
`;
